src/format/parse.rs

```diff
@@ -1040,3 +1298,135 @@ fn test_issue_1010() {
+    assert_eq!(parse("+2023-W23-5 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+02023-W23-5 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+12023-W23-5 "), Ok((from_isoywd_opt(12023, 23, Fri).unwrap(), " ")));
+    // mixed basic and extended format
+    assert_eq!(parse("2023-W235 "), Err(INVALID));
+    assert_eq!(parse("202306-W235 "), Err(INVALID));
+    assert_eq!(parse("-2023-W235 "), Err(INVALID));
+    // No padding
+    assert_eq!(parse("2023-W25 "), Err(INVALID));
+    assert_eq!(parse("23-W23-5 "), Err(INVALID));
+    // Year is interpreted as `iso_year`
+    assert_eq!(parse("2022-W52-7 "), Ok((NaiveDate::from_ymd_opt(2023, 1, 1).unwrap(), " ")));
+}
+
+#[cfg(test)]
+#[test]
+fn test_parse_iso8601_time() {
+    fn parse(s: &str) -> ParseResult<(&str, u32, u32, u32, u32, bool)> {
+        let mut parsed = Parsed::new();
+        let (s, _, hour24) = parse_iso8601_time(&mut parsed, s)?;
+        Ok((
+            s,
+            12 * parsed.hour_div_12.unwrap() + parsed.hour_mod_12.unwrap(),
+            parsed.minute.unwrap(),
+            parsed.second.unwrap_or(0),
+            parsed.nanosecond.unwrap_or(0),
+            hour24,
+        ))
+    }
+
+    // basic format, complete representation
+    assert_eq!(parse("152830 "), Ok((" ", 15, 28, 30, 0, false)));
+    // extended format, complete representation
+    assert_eq!(parse("15:28:30 "), Ok((" ", 15, 28, 30, 0, false)));
+    // basic format, fractional second
+    assert_eq!(parse("152830,6 "), Ok((" ", 15, 28, 30, 600_000_000, false)));
+    assert_eq!(parse("152830.60 "), Ok((" ", 15, 28, 30, 600_000_000, false)));
+    // extended format, fractional second
+    assert_eq!(parse("15:28:30,6 "), Ok((" ", 15, 28, 30, 600_000_000, false)));
+    assert_eq!(parse("15:28:30.60 "), Ok((" ", 15, 28, 30, 600_000_000, false)));
+    // basic format, fractional minute
+    assert_eq!(parse("1528,5 "), Ok((" ", 15, 28, 30, 0, false)));
+    assert_eq!(parse("1528.51 "), Ok((" ", 15, 28, 30, 600_000_000, false)));
+    // extended format, fractional minute
+    assert_eq!(parse("15:28,5 "), Ok((" ", 15, 28, 30, 0, false)));
+    assert_eq!(parse("15:28.51 "), Ok((" ", 15, 28, 30, 600_000_000, false)));
+    // extended format, fractional hour
+    assert_eq!(parse("15,45 "), Ok((" ", 15, 27, 0, 0, false)));
+    assert_eq!(parse("15.12345 "), Ok((" ", 15, 7, 24, 420_000_000, false)));
+    assert_eq!(parse("15,99999999999999999999 "), Ok((" ", 15, 59, 59, 999_999_999, false)));
+
+    // 24:00:00 is allowed
+    assert_eq!(parse("240000 "), Ok((" ", 0, 0, 0, 0, true)));
+    assert_eq!(parse("24:00:00 "), Ok((" ", 0, 0, 0, 0, true)));
+    assert_eq!(parse("24:00:00,0 "), Ok((" ", 0, 0, 0, 0, true)));
+    // But no times beyond that
+    assert_eq!(parse("24:30:00 "), Err(INVALID));
+    assert_eq!(parse("24:00:30 "), Err(INVALID));
+    assert_eq!(parse("24:00:00,5 "), Err(INVALID));
+
```

Comment:
Recommend some variations of large unicode characters to check for splitting unicode within the char (nothign should panic).

     assert_eq!(parse("2💀 "), Err(INVALID));
     assert_eq!(parse("24💀 "), Err(INVALID));
     assert_eq!(parse("240💀 "), Err(INVALID));
     assert_eq!(parse("2400💀 "), Err(INVALID));
     assert_eq!(parse("24000💀 "), Err(INVALID));
     assert_eq!(parse("240000💀 "), Err(INVALID));
     assert_eq!(parse("240000.💀 "), Err(INVALID));
     assert_eq!(parse("240000.5💀 "), Err(INVALID));